### DISCUSSION TITLE
Incorrect Code

### DISCUSSION THREAD

**Discussion by ttarikyuksel** [Authenticator-Extension/Authenticator#1053] [General]
*************************************************************************************************************************************************************************************************************************************************************************

**Comment by Sneezry** [comment]
There are two possible reasons for an incorrect code despite a correct secret. The first reason may be that the clock is incorrect. To resolve this issue, you can synchronize your clock with an NTP server. The second reason could be that the algorithm selected does not match the one used on the server side. We support SHA-1, SHA-256, SHA-512, GOST 34.11 256, and GOST 34.11 512. By default, SHA-1 is selected if no algorithm is specified when adding a secret. If you need to change the algorithm, you can do so by accessing the advanced options when adding a new account.

![image](https://github.com/Authenticator-Extension/Authenticator/assets/1621293/bf1c89b2-0191-4196-b6a1-60a6e1a73ba7)

**Reply by ttarikyuksel** [reply]
Thank you for your quick action, I will try this.
